stdlib/ballerina-http/src/main/java/org/ballerinalang/net/uri/URIUtil.java

```diff
@@ -39,4 +44,35 @@ public static String getSubPath(String path, String basePath) {
 
         return path.substring(basePath.length());
     }
+
+    public static String removeMatrixParams(String path, Map<String, Map<String, String>> matrixParams) {
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+        String[] pathSplits = path.split("\\?");
+        String[] pathSegments = pathSplits[0].split("/");
+        String pathToMatrixParam = "";
+        for (String pathSegment : pathSegments) {
+            String[] splitPathSegment = pathSegment.split(";");
+            pathToMatrixParam = pathToMatrixParam.concat("/" + splitPathSegment[0]);
+            Map<String, String> segmentMatrixParams = new HashMap<>();
+            if (splitPathSegment.length > 1) {
+                for (int i = 1; i < splitPathSegment.length; i++) {
+                    String[] splitMatrixParam = splitPathSegment[i].split("=");
+                    if (splitMatrixParam.length != 2) {
+                        throw new BallerinaConnectorException("Found non matrix parameter in " + path);
```

Comment:
It would be better to include non-matrix param itself in the message. ie:
`throw new BallerinaConnectorException("Found non matrix parameter " + param + " in " + path)`